data_steward/resources.py

```diff
@@ -153,27 +196,31 @@ def cdm_schemas(include_achilles=False, include_vocabulary=False):
     :return:
     """
     result = dict()
-    for f in os.listdir(fields_path):
-        file_path = os.path.join(fields_path, f)
-        with open(file_path, 'r') as fp:
-            file_name = os.path.basename(f)
-            table_name = file_name.split('.')[0]
-            schema = json.load(fp)
-            include_table = True
-            if table_name in VOCABULARY_TABLES and not include_vocabulary:
-                include_table = False
-            elif table_name in ACHILLES_TABLES + ACHILLES_HEEL_TABLES and not include_achilles:
-                include_table = False
-            elif is_internal_table(table_name):
-                include_table = False
-            elif is_pii_table(table_name):
-                include_table = False
-            elif is_id_match(table_name):
-                include_table = False
-            elif table_name == 'post_deid_person':
-                include_table = False
-            if include_table:
-                result[table_name] = schema
+    for dir_path, _, files in os.walk(fields_path):
```

Comment:
This works with the current state of `resource_files` but doesn't do the same name collision check as `fields_for` does; If I remember correctly and `os.walk` does do a depth-first search then a theoretical file `resource_files/post_deid/person.json` would thereby get returned rather than `resource_files/person.json`. Please let me know if I'm mistaken.

Even still we might want to account for possible collisions. Could we and would it help to put the standard schemas (clinical + vocabulary + achilles + heel) in `resource_files/omop`? If it's scope creepy, then maybe we could just raise an error on name collision here until such time that we can reorg all schemas into folders as we'd like.

Reply:
Yes, this will need to be adjusted, but that does feel like scope creep.  I think this is an engineering improvement and another ticket.

And os.walk does use a depth first search.  But is defaults to `topdown=True`, meaning it will find files higher in the directory system first.

Reply:
I built a throw away directory structure like:
```
temp/
temp/person.json
temp/alpha/person.json
temp/test/person.json
```
I then ran the following in an interpreter and got the following results.
```
>>> import os
>>> for dir_path, _, files in os.walk('temp'):
...   print(f"{dir_path}\t\t\t{files}")
... 
temp			['person.json']
temp/test			['person.json']
temp/alpha			['person.json']
>>> for dir_path, _, files in os.walk('temp', topdown=False):
...   print(f"{dir_path}\t\t\t{files}")
... 
temp/test			['person.json']
temp/alpha			['person.json']
temp			['person.json']
>>> for dir_path, _, files in os.walk('temp', topdown=True):
...   print(f"{dir_path}\t\t\t{files}")
... 
temp			['person.json']
temp/test			['person.json']
temp/alpha			['person.json']
```
It is looping through the files in a topdown directory order by default.

Reply:
Got it @lrwb-aou. It makes sense to include this as part of a bigger folder re-org ticket. And I think this is ok since walking would resolve to the existing schemas.

Reply:
I definitely want to do a folder re-org.  It would ease multiple areas.  But I just think that is an engineering improvement this ticket doesn't especially need.

Reply:
Created DC-1015 to address this.